src/libratbag-data.c

```diff
@@ -269,6 +283,98 @@ init_data_steelseries(struct ratbag *ratbag,
 	}
 }
 
+static void
+init_data_asus(struct ratbag *ratbag,
+		GKeyFile *keyfile,
+		struct ratbag_device_data *data)
+{
+	const char *group = "Driver/asus";
+	int num, i;
+	GError *error = NULL;
+	_cleanup_(freep) char *str = NULL;
+	_cleanup_(g_strfreevp) char **strv = NULL;
+
+	data->asus.profile_count = -1;
+	data->asus.button_count = -1;
+	data->asus.led_count = -1;
+	data->asus.dpi_count = -1;
+	data->asus.dpi_range = NULL;
+	data->asus.is_wireless = -1;
+	data->asus.quirks = 0;
+	for (i = 0; i < ASUS_MAX_NUM_BUTTON; i++)
+		data->asus.button_mapping[i] = -1;
+
+	error = NULL;
+	num = g_key_file_get_integer(keyfile, group, "Profiles", &error);
+	if (!error && num >= 0)
+		data->asus.profile_count = num;
+	if (error)
+		g_error_free(error);
+
+	error = NULL;
+	num = g_key_file_get_integer(keyfile, group, "Buttons", &error);
+	if (!error && num >= 0 && num <= ASUS_MAX_NUM_BUTTON)
+		data->asus.button_count = num;
+	if (error)
+		g_error_free(error);
+
+	error = NULL;
+	strv = g_key_file_get_string_list(keyfile, group, "ButtonMapping", NULL, NULL);
+	if (strv) {
+		i = 0;
+		while (strv[i]) {
+			data->asus.button_mapping[i] = (int8_t) strtoul(strv[i], NULL, 10);
+			i++;
+		}
```

Comment:
```suggestion
		for (unsigned int i = 0; i <= list_length; i++) {
			data->asus.button_mapping[i] = (int8_t) strtoul(strv[i], NULL, 10);
		}
```
Please, also check if my suggested range `i <= list_length` is correct.